### PR TITLE
React select styling - unsafe-inline issue

### PR DESCRIPTION
**Pull request by timarney**
# Summary | Résumé

Fixes: https://github.com/cds-snc/platform-forms-client/issues/3352

Current on preview environment. 

screenshot_2024-04-25_at_2 25 08___pm

<hr>

## Notes

With our CSP script blocking `unsafe-inline` the react select library styles are getting blocked.

There is an option to set an `unstyled` property however some base styles still try to load and are currently needed.

> If you are trying to style everything from scratch you can use the unstyled prop. This removes all the presentational styles from React Select (leaving some important functional styles, like those for menu positioning and input width in multi select).

https://react-select.com/styles#the-unstyled-prop

There is a closed issue on the repo discussing the issue here:

https://github.com/JedWatson/react-select/issues/2706

## Fix 

Screenshot 2024-04-26 at 11 47 46 AM

This PR adds `unstyled` prop & uses Tailwind for most of the styling however as noted above inline styles are still required.

👉  To fix the inline styles issue the CSP has been updated for now to allow inline styles when a cookie session exists

### CODE REVIEW

**Comment by timarney** [comment]
Might be possible to use 

https://emotion.sh/docs/@emotion/cache#nonce-string

**Comment by timarney** [comment]
Updated to use Emotion Cache key to provide a nonce and avoid unsafe inline


https://github.com/cds-snc/platform-forms-client/pull/3521/files#diff-6d7e944d154e9748f30bffc299f0627af3c9d572a96b837b438d0701b6ac3515R46

File: app/(gcforms)/[locale]/(form administration)/form-builder/[id]/settings/manage/FormOwnership.tsx [hunk 1]
```diff
@@ -1,28 +1,25 @@
...
 import { logMessage } from "@lib/logger";
 import { Button } from "@clientComponents/globals";
-import Select from "react-select";
-import makeAnimated from "react-select/animated";
+import { FormOwnerSelect, usersToOptions } from "./FormOwnerSelect";
+
+import { CacheProvider, EmotionCache } from "@emotion/react";
+import createCache from "@emotion/cache";
+
+// import makeAnimated from "react-select/animated";
```
Comment (bryan-robitaille) [inline app/(gcforms)/[locale]/(form administration)/form-builder/[id]/settings/manage/FormOwnership.tsx:12]: Assuming commenting out the import is left over debugging?

```suggestion
```

File: lib/cspScripts.ts [hunk 2]
```diff
@@ -45,7 +46,7 @@ export const generateCSP = (): { csp: string; nonce: string } => {
   const cspHeader = `
     default-src 'self';
     script-src 'self' 'nonce-${nonce}' 'strict-dynamic';
-    style-src 'self' 'nonce-${nonce}';
+    style-src 'self' ${styleSrc};
```
Comment (bryan-robitaille) [inline lib/cspScripts.ts:49]: Curious why the change in value to a new declared variable?
```suggestion
    style-src 'self' 'nonce-${nonce}'
```

File: lib/cspScripts.ts [hunk 3]
```diff
@@ -24,8 +24,9 @@ if (window.location.host === "forms-formulaires.alpha.canada.ca") {
 }
 `;
 
-export const generateCSP = (): { csp: string; nonce: string } => {
+export const generateCSP = () => {
   const nonce = Buffer.from(crypto.randomUUID()).toString("base64");
+  const styleSrc = `'nonce-${nonce}'`;
```
Comment (bryan-robitaille) [inline lib/cspScripts.ts:29]: Might be better to leave inline?
```suggestion
```

File: lib/cspScripts.ts [hunk 4]
```diff
@@ -24,8 +24,9 @@ if (window.location.host === "forms-formulaires.alpha.canada.ca") {
 }
 `;
 
-export const generateCSP = (): { csp: string; nonce: string } => {
```
Comment (bryan-robitaille) [inline lib/cspScripts.ts:27]: Did we run into an issue further in the PR with the typing?

**Review by bryan-robitaille — CHANGES_REQUESTED** [review]
A little left over cleanup after doing amazing work to get the Emotion lib to use nonce's.

**Review by bryan-robitaille — DISMISSED** [review]
Nice 👍

**Review by dsamojlenko — APPROVED** [review]
